dts/bindings/adc/st,stm32-adc.yaml

```diff
@@ -58,5 +58,18 @@ properties:
       By design, these macros also contains all register information (address,
       field offset and field mask) necessary to properly set the resolution.
 
+  sampling-times:
+    type: array
+    required: true
+    description: |
+      List all the supported sampling time supported by the ADC instance.
+      Rounded up if needed. Order is important: their index matches their binary
+      value in the register.
+
+  sampling-time-common-channel-nb:
+    type: int
+    description: |
+      Number of sampling time common channel for this ADC instance, if any.
```

Comment:
I hate making comments about variable/property names.  But this one really doesn't look like it follows the existing pattern.

A search of all property names in Zephyr bindings, and also the the much larger Linux bindings set, shows that usually properties that indicate the number or count of items are named "num-thing" e.g. `num-sampling-time-common-channels`.  Like in `num-cs`, `num-channels`, etc that already exist.

A much smaller set use the pattern "thing-count".

I couldn't find any existing properties using the pattern "*-nb" as a name.

In Linux, I found 15 unique property names with the pattern "*-count" and 72 with the pattern "num-*".  The latter are used 136 times, as many different drivers use, e.g., `num-cs` and `num-ports`.  The "*-count" properties are all specific to a single driver defining them and not re-used.  Also note that most commonly the thing is plural, i.e. `num-channels` and not `num-channel`.

Also, it would also be more correct to say, "Number of sampling time common channel**s** …"  That is, make "channels" plural.  Normally I wouldn't nitpick English grammar, but here it is ambiguous if this should mean, "Index of sampling time common channel …"    _Number_ can, in common usage, mean either _index_ or _count_.  A programmer would usually prefer a more precise term like _index_ vs using _number_ to mean _index_.  But meaning _count_, the noun should be plural, while meaning _index_, the noun should be singular.  So I found it ambiguous if this was a trivial grammar error or it meant the index of a specific channel to use for the common sampling time.

Reply:
No problem about this comment. I admit I didn't search what the usual convention is. I'll change to `num-sampling-time-common-channels` which is still very long, but I don't see a good way to reduce the size without losing meaning.